src/Header/Header.spec.tsx

```diff
@@ -0,0 +1,84 @@
+import { screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { setCalendarDate } from '../store/slices/calendarDateSlice';
+import { renderWithProviders } from '../utils/testUtils';
+import Header from './Header';
+import { addDays } from '../dateUtils';
+
+const normalizeDate = (date: Date) => {
+  const normalized = new Date(date);
+  normalized.setMilliseconds(0);
+  normalized.setSeconds(0);
+  return normalized;
+};
+
+describe('Header Component', () => {
+  describe('when initially rendered', () => {
+    it('should display the calendar title', () => {
+      renderWithProviders(<Header />);
+
+      expect(screen.getByText(/Calendar/)).toBeInTheDocument();
+    });
+
+    it('should have the initial state of calendar date', () => {
+      const { store } = renderWithProviders(<Header />);
+
+      const initialState = store.getState().calendar.calendarDate;
+      expect(initialState).toBeInstanceOf(Date);
```

Comment:
These tests should be testing the component, here you check only the redux state. Remove the test.
```suggestion
      const { store } = renderWithProviders(<Header />);

      const initialState = store.getState().calendar.calendarDate;
      expect(initialState).toBeInstanceOf(Date);
```